pkg/util/node/node.go

```diff
@@ -170,187 +165,187 @@ func GetNodeIP(client clientset.Interface, name string) net.IP {
-	oldData, err := json.Marshal(oldNode)
-	if err != nil {
-		return nil, fmt.Errorf("failed to Marshal oldData for node %q: %v", nodeName, err)
-	}
-
-	// NodeStatus.Addresses is incorrectly annotated as patchStrategy=merge, which
-	// will cause strategicpatch.CreateTwoWayMergePatch to create an incorrect patch
-	// if it changed.
-	manuallyPatchAddresses := (len(oldNode.Status.Addresses) > 0) && !equality.Semantic.DeepEqual(oldNode.Status.Addresses, newNode.Status.Addresses)
-
-	// Reset spec to make sure only patch for Status or ObjectMeta is generated.
-	// Note that we don't reset ObjectMeta here, because:
-	// 1. This aligns with Nodes().UpdateStatus().
-	// 2. Some component does use this to update node annotations.
-	diffNode := newNode.DeepCopy()
-	diffNode.Spec = oldNode.Spec
-	if manuallyPatchAddresses {
-		diffNode.Status.Addresses = oldNode.Status.Addresses
-	}
-	newData, err := json.Marshal(diffNode)
-	if err != nil {
-		return nil, fmt.Errorf("failed to Marshal newData for node %q: %v", nodeName, err)
-	}
-
-	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to CreateTwoWayMergePatch for node %q: %v", nodeName, err)
-	}
-	if manuallyPatchAddresses {
-		patchBytes, err = fixupPatchForNodeStatusAddresses(patchBytes, newNode.Status.Addresses)
-		if err != nil {
-			return nil, fmt.Errorf("failed to fix up NodeAddresses in patch for node %q: %v", nodeName, err)
-		}
-	}
-
-	return patchBytes, nil
-}
-
-// fixupPatchForNodeStatusAddresses adds a replace-strategy patch for Status.Addresses to
-// the existing patch
-func fixupPatchForNodeStatusAddresses(patchBytes []byte, addresses []v1.NodeAddress) ([]byte, error) {
-	// Given patchBytes='{"status": {"conditions": [ ... ], "phase": ...}}' and
-	// addresses=[{"type": "InternalIP", "address": "10.0.0.1"}], we need to generate:
-	//
-	//   {
-	//     "status": {
-	//       "conditions": [ ... ],
-	//       "phase": ...,
-	//       "addresses": [
-	//         {
-	//           "type": "InternalIP",
-	//           "address": "10.0.0.1"
-	//         },
-	//         {
-	//           "$patch": "replace"
-	//         }
-	//       ]
-	//     }
-	//   }
-
-	var patchMap map[string]interface{}
-	if err := json.Unmarshal(patchBytes, &patchMap); err != nil {
-		return nil, err
-	}
-
-	addrBytes, err := json.Marshal(addresses)
-	if err != nil {
-		return nil, err
-	}
-	var addrArray []interface{}
-	if err := json.Unmarshal(addrBytes, &addrArray); err != nil {
-		return nil, err
-	}
-	addrArray = append(addrArray, map[string]interface{}{"$patch": "replace"})
-
-	status := patchMap["status"]
-	if status == nil {
-		status = map[string]interface{}{}
-		patchMap["status"] = status
-	}
-	statusMap, ok := status.(map[string]interface{})
-	if !ok {
-		return nil, fmt.Errorf("unexpected data in patch")
-	}
-	statusMap["addresses"] = addrArray
-
-	return json.Marshal(patchMap)
-}
+// type nodeForConditionPatch struct {
```

Comment:
This code needs moving to the new location (also then please delete this package, rather than commenting it out)

Reply:
Done